packages/graphql/src/services/questions.ts

```diff
@@ -363,3 +343,74 @@ export async function toggleIsArchived(
     isArchived,
   }))
 }
+
+// map mime types of images to file extensions
+const FILE_EXTENSIONS: Record<string, string> = {
+  'image/png': 'png',
+  'image/jpeg': 'jpg',
+  'image/gif': 'gif',
+  'image/svg+xml': 'svg',
+  'image/webp': 'webp',
+  'image/tiff': 'tiff',
+  'image/bmp': 'bmp',
+}
+
+export async function getFileUploadSas(
+  { fileName, contentType }: { fileName: string; contentType: string },
+  ctx: ContextWithUser
+) {
+  const sharedKeyCredential = new StorageSharedKeyCredential(
+    process.env.BLOB_STORAGE_ACCOUNT_NAME as string,
+    process.env.BLOB_STORAGE_ACCESS_KEY as string
+  )
+
+  const storageAccount = `https://${
+    process.env.BLOB_STORAGE_ACCOUNT_NAME as string
+  }.blob.core.windows.net`
+
+  // if nonexistent, create a container for the user on blob storage
+  const client = new BlobServiceClient(storageAccount, sharedKeyCredential)
+  const containerClient = client.getContainerClient(ctx.user.sub)
```

Comment:
each user gets a container in blob storage with the user id. the generated token allows for a file upload to exactly one location as specified by the randomly generated UUID.

Reply:
the migration will put all of the existing images into this container as well